instrumentation/grizzly-client-1.9/src/test/groovy/GrizzlyAsyncHttpClientTest.groovy

```diff
@@ -20,10 +20,13 @@ import com.ning.http.client.Request
 import com.ning.http.client.RequestBuilder
 import com.ning.http.client.Response
 import com.ning.http.client.uri.Uri
+import io.opentelemetry.auto.test.SpockRunner
 import io.opentelemetry.auto.test.base.HttpClientTest
+import org.junit.runner.RunWith
 import spock.lang.AutoCleanup
 import spock.lang.Shared
 
+@RunWith(SpockRunner)
```

Comment:
When I thought I finally had a good mental model of what's going on, a grizzly shows up 😅 

I think one of the reasons a couple of test failed is because with JUnit5, there is no way to intercept class loading
https://github.com/junit-team/junit5/issues/201

This is a fundamental aspect of JUnit5 so presumably even when using the vintage engine, the limitation applies.

I noticed only two tests in the entire repo actually seem to rely on this behavior though, so I gave them special treatment. At that point, it seemed `SpockRunner` doesn't actually do anything anymore, so I tried removing it completely. Everything worked, except for this one test. 

@trask if you feel like it, maybe you could look at this test without and with SpockRunner and teach me what function it's serving and why it would only affect this test :)

Reply:
Yeah, that's a tricky one...

There are two things conspiring against that test:

1. the test code itself gets instrumented, not only the library under test (`GrizzlyAsyncHttpClientTest$AsyncCompletionHandlerMock`)
2. the instrumentation that gets injected into that test code uses `InstrumentationContext`, and tries to put a `null` value into it

Because `AsyncCompletionHandlerMock` is a nested class inside of the test, it appears that class is loaded before we install bytebuddy, which means we can only retransform it, which means `InstrumentationContext` cannot use the (fast) field-backed provider, and instead has to use the `WeakMap`.

This is fine.

But then comes (2) and tries to put a null value into the `WeakMap`, which is what then causes the NPE.

Moving `AsyncCompletionHandlerMock` out into a top-level class resolves the issue.

I'll also open an issue about the `WeakMap` implementation not supporting `null` values, though in general we don't use the the `WeakMap` when running as `-javaagent:...`, since in that case we are typically able to do initial transform (not structure limited re-transform) and so we can inject a field into the class and use the field-backed provider.